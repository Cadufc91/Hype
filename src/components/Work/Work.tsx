import * as React from 'react';
import { 
    WorkContainer,
    WorkTitle,
    Divider1,
    WorkCards,
    CardItem,
    CardImage,
    CardInfo,
    CardTitle,
    ProjectLink
} from './style';
import Learning from '../../assets/img/learning1.png';
import Learning2 from '../../assets/img/learning2.png';
import Marketing from '../../assets/img/marketing.png';
import Blog from '../../assets/img/blog.png';
import Educare from '../../assets/img/educare.png';
import Blogy from '../../assets/img/blogy.png';
import { BsArrowRight } from 'react-icons/bs';

const Work = () => {
    return(
        <WorkContainer>
            <WorkTitle>My Selected Work</WorkTitle>
            <Divider1 />
            <WorkCards>
                <CardItem>
                    <CardImage src={Learning} alt="" />
                    <CardInfo>
                        <CardTitle>Eduguard - E-learning Website</CardTitle>
                        <ProjectLink>View Project</ProjectLink>
                        <BsArrowRight size={20}/>
                    </CardInfo>
                </CardItem>
                <CardItem>
                    <CardImage src={Learning2} alt="" />
                    <CardInfo>
                        <CardTitle>Eduguard E-learning Website</CardTitle>
                        <ProjectLink>View Project</ProjectLink>
                        <BsArrowRight size={20}/>

                    </CardInfo>
                </CardItem>
                <CardItem>
                    <CardImage src={Marketing} alt="" />
                    <CardInfo>
                        <CardTitle>Huma Marketing Agency</CardTitle>
                        <ProjectLink>View Project</ProjectLink>
                        <BsArrowRight size={20}/>

                    </CardInfo>
                </CardItem>
                <CardItem>
                    <CardImage src={Blog} alt="" />
                    <CardInfo>
                        <CardTitle>Beeryblog Blog Website</CardTitle>
                        <ProjectLink>View Project</ProjectLink>
                        <BsArrowRight size={20} />

                    </CardInfo>
                </CardItem>
                <CardItem>
                    <CardImage src={Educare} alt="" />
                    <CardInfo>
                        <CardTitle>Educare Elearning website</CardTitle>
                        <ProjectLink>View Project</ProjectLink>
                        <BsArrowRight size={20}/>

                    </CardInfo>
                </CardItem>
                <CardItem>
                    <CardImage src={Blogy} alt="" />
                    <CardInfo>
                        <CardTitle>Blogy Blog Website</CardTitle>
                        <ProjectLink>View Project</ProjectLink>
                        <BsArrowRight size={20}/>

                    </CardInfo>
                </CardItem>
            </WorkCards>
        </WorkContainer>
    )
};

export default Work;