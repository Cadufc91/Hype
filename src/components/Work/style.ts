import styled from "styled-components";

export const WorkContainer = styled.section`
    height: 1247px;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 100px;
    max-width: 70%;
    margin-left: 15%;
`;

export const WorkTitle = styled.h1`
    font-size: 40px;
    font-weight: 400;
    color: #fff
`;

export const Divider1 = styled.span`
    width: 80px;
    height: 3px;
    background: #ED1CA6;
    border: 1px solid #ED1CA6;
    margin-bottom: 50px;
`;

export const WorkCards = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
`;

export const CardItem = styled.div`
    background: #fff;
    margin-left: 24px;
    margin-bottom: 24px;
    border-radius: 8px;
    max-width: 30%;
    width: 300px;
`;

export const CardImage = styled.img`
    width: 100%;
    height: 150px;
`;

export const CardInfo = styled.div`
    padding: 24px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`;

export const CardTitle = styled.h2`
    font-size: 20px;
    cursor: pointer;

    a{
        text-decoration: none;
        color: #000;

        :hover{
            color: #ED1CA6;
        }
    }
`;

export const ProjectLink = styled.p`
    display: none;
`;

