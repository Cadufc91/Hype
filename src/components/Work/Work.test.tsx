import { render } from "@testing-library/react";
import  { Work } from './Work';

describe('Header', () => {
    it('should match snapshot', () => {
        const { container } = render(<Work />)

        expect(container.firstChild).toMatchSnapshot()
    })
})