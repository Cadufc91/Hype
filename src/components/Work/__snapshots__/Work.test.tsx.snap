// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<section
  class="sc-bczRLJ yKxNI"
  id="work"
>
  <h1
    class="sc-gsnTZi dnxmsr"
  >
    Meus Trabalhos
  </h1>
  <span
    class="sc-dkzDqf jdJdDv"
  />
  <div
    class="sc-hKMtZM iXqeDZ"
  >
    <div
      class="sc-eCYdqJ jZNbkl"
    >
      <img
        alt=""
        class="sc-jSMfEi cEPJYg"
        src="proj1.jpg"
      />
      <div
        class="sc-gKXOVf hXmtcq"
      >
        <h2
          class="sc-iBkjds bnjcEE"
        >
          <a
            href="https://github.com/Cadufc91/SPA"
          >
            Site para um SPA
          </a>
        </h2>
        <p
          class="sc-ftvSup jDzSUa"
        >
          Ver Projeto
        </p>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jZNbkl"
    >
      <img
        alt=""
        class="sc-jSMfEi cEPJYg"
        src="proj2.jpg"
      />
      <div
        class="sc-gKXOVf hXmtcq"
      >
        <h2
          class="sc-iBkjds bnjcEE"
        >
          <a
            href="https://github.com/Cadufc91/controle-financeiro"
          >
             Controle Financeiro
          </a>
        </h2>
        <p
          class="sc-ftvSup jDzSUa"
        >
          View Project
        </p>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jZNbkl"
    >
      <img
        alt=""
        class="sc-jSMfEi cEPJYg"
        src="proj3.jpg"
      />
      <div
        class="sc-gKXOVf hXmtcq"
      >
        <h2
          class="sc-iBkjds bnjcEE"
        >
          <a
            href="https://github.com/Cadufc91/Plan-teste"
          >
            Site desenvolvedora de software
          </a>
           
        </h2>
        <p
          class="sc-ftvSup jDzSUa"
        >
          View Project
        </p>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jZNbkl"
    >
      <img
        alt=""
        class="sc-jSMfEi cEPJYg"
        src="proj4.jpg"
      />
      <div
        class="sc-gKXOVf hXmtcq"
      >
        <h2
          class="sc-iBkjds bnjcEE"
        >
          <a
            href="https://github.com/Cadufc91/Alura-Geek"
          >
            E-commerce de games e consoles
          </a>
           
        </h2>
        <p
          class="sc-ftvSup jDzSUa"
        >
          View Project
        </p>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jZNbkl"
    >
      <img
        alt=""
        class="sc-jSMfEi cEPJYg"
        src="proj5.jpg"
      />
      <div
        class="sc-gKXOVf hXmtcq"
      >
        <h2
          class="sc-iBkjds bnjcEE"
        >
          <a
            href="https://github.com/Cadufc91"
          >
            Landing page para captação de leads
          </a>
           
        </h2>
        <p
          class="sc-ftvSup jDzSUa"
        >
          View Project
        </p>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jZNbkl"
    >
      <img
        alt=""
        class="sc-jSMfEi cEPJYg"
        src="proj6.jpg"
      />
      <div
        class="sc-gKXOVf hXmtcq"
      >
        <h2
          class="sc-iBkjds bnjcEE"
        >
          <a
            href="https://github.com/Cadufc91/Desafio-H"
          >
            Landing page para venda de produtos
          </a>
        </h2>
        <p
          class="sc-ftvSup jDzSUa"
        >
          View Project
        </p>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</section>
`;
